lib/game_core.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module IcodebreakerGem
+  class GameCore
+    include Validation
+
+    def initialize(secret)
+      validate_code secret
+
+      @secret = secret.to_s
+      @shuffle = @secret.chars.shuffle
+    end
+
+    def compare(guess)
+      return '++++' if @secret == guess
+      return '' if (@secret.chars & guess.chars).blank?
```

Comment:
my bad, method `blank?` provided by `rails` not by `ruby`
move back `empty?` instead